purekit/src/main/kotlin/com/virgilsecurity/purekit/pure/VirgilCloudPureStorage.java

```diff
@@ -125,9 +173,27 @@ public UserRecord selectUser(String userId)
         return userRecord;
     }
 
+    /**
+     * Obtains a users record with the given userId from a storage.
+     *
+     * @param userIds User Ids. Should not contain duplicates.
+     *
+     * @return UserRecords.
+     *
+     * @throws PureException If user Id duplicate has been found or user id mismatches the one
+     * from a server. Use {@link PureException#getErrorStatus()} to know the particular case.
+     * @throws ProtocolException Thrown if an error from the PHE service has been parsed
+     * successfully.
+     * @throws ProtocolHttpException Thrown if an error from the PHE service has NOT been parsed
+     * successfully. Represents a regular HTTP exception with code and message.
+     * @throws InvalidProtocolBufferException If a PurekitProtosV3Storage.UserRecord received from
+     * a server cannot be parsed as a Protobuf message.
+     * @throws VerificationException If signature verification operation failed.
+     */
     @Override
-    public Iterable<UserRecord> selectUsers(Collection<String> userIds) // FIXME maybe just use Set instead of Iterable? so we can avoid DUPLICATE_USER_ID
-        throws PureException, ProtocolException, ProtocolHttpException {
+    public Iterable<UserRecord> selectUsers(Collection<String> userIds) // FIXME maybe just use Set instead of Collection? so we can avoid DUPLICATE_USER_ID
```

Comment:
Remove fix

Reply:
what's about Set?

Reply:
Do you now storage that returns collection of entities in form of Set? I don't. And if there is no such storages, fetching all entities and then pushing them to Set if overhead

Reply:
Collection<String> userIds -> Set<String>
it's argument, not return type

Reply:
I though we're talking about return type. Argument may be changed to Set, that doesn't matter so much